packages/server/lib/util/ensure-url.ts

```diff
@@ -0,0 +1,24 @@
+import * as url from 'url'
+import * as rp from 'request-promise'
+import { agent, connect } from '@packages/networking'
+
+export function ensureUrl (urlStr: string) {
```

Comment:
maybe you can rename this to be `isListening`

`ensureUrl.isListening(...)` would be the call signature so it reads a bit better